Merge pull request #240 from krlmlr/f-examine-to-check (#240)
krlmlr/dm: vignettes/wip/dm-function-naming-logic.Rmd

1. `dm_`: There are two cases for this prefix:
    - The function takes a `dm` object as a first argument
    - The function constructs a `dm` object from other objects
1. `check_`: Certain tests on the arguments are performed and an error is thrown in case they are not passed. If the tests are passed, the first argument is returned invisibly.

> 1. `examine_`: The function executes checks (of which some may lead to an error if not passed) and returns an object containing an overview of the results of the checks. 

Reviewer: ```suggestion
1. `examine_`: The function executes checks and returns an object containing an overview of the results of the checks. It should never throw an error.
```

Reviewer: Do we still need to rephrase here?
